### PR TITLE
New Workaround for mergeable_state Github bug

### PR DESCRIPTION
**Pull request by sileht**

### CODE REVIEW

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 75315b1661aaf20857ea90eaf00c9b468a66e0d1

**Comment by sileht** [comment]
Pull-request updated, HEAD is now f46f965bae668cc7a40435d790dc040680c34548

File: mergify_engine/gh_pr_fullifier.py [hunk 1]
```diff
@@ -162,6 +162,7 @@ def disabled_by_rules(pull, **extra):
 # unstable: branch up2date (if strict: True) and not required status
 #           checks are failure or pending
 # clean: branch up2date (if strict: True) and all status check OK
+# has_hooks: Mergeable with passing commit status and pre-recieve hooks.
```
Comment (jd) [inline mergify_engine/gh_pr_fullifier.py]: receive

File: mergify_engine/gh_pr_fullifier.py [hunk 2]
```diff
@@ -179,62 +180,66 @@ class MergifyState(object):
...
     disabled = disabled_by_rules(pull, **extra)
 
-    # TODO(sileht): remove me
-    if len(pull.mergify_engine["approvals"]) == 4:
-        reviews_ok, reviews_ko, reviews_required, _ = \
-            pull.mergify_engine["approvals"]
-    else:
-        reviews_ok, reviews_ko, reviews_required = \
-            pull.mergify_engine["approvals"]
+    reviews_ok, reviews_ko, reviews_required = pull.mergify_engine["approvals"]
```
Comment (jd) [inline mergify_engine/gh_pr_fullifier.py]: looks unrelated but OK :)

**Pull request merged**